### PR TITLE
fix(guest-portal): read/write setting/guest_access instead of a fictional endpoint

### PR DESCRIPTION
**Pull request by jpennin5**
## Description

**Problem:** Both guest portal tools call
`/integration/v1/sites/{site}/guest-portal/config`, an endpoint no Network
version serves. Every call fails with an HTTP 404 on every controller, so
`get_guest_portal_config` and `configure_guest_portal` have never worked.

**Root cause:** The endpoint and the `GuestPortalConfig` model describe an
imagined API. Guest portal configuration actually lives in the legacy
`setting/guest_access` section (`get/setting/guest_access` /
`set/setting/guest_access/{id}`).

**Fix:**

- `get_guest_portal_config` reads the section and translates it: the
  section reports `auth: "hotspot"` for several portal modes, so the mode
  is disambiguated by the `password_enabled` / `voucher_enabled` /
  `radius_enabled` flags. The raw section is passed through with `x_*`
  secret material stripped.
- `configure_guest_portal` gains `portal_enabled` — the switch that turns
  the captive portal off entirely while `purpose=guest` networks keep
  their isolation. Writes are read-modify-write against the section `_id`;
  selecting an auth method clears the sibling `*_enabled` flags;
  portal-customization keys are written only if this controller version
  reports them, with anything else named in `skipped_fields`; the tool
  re-reads when the controller does not echo the PUT.
- Response unwrapping uses a module-local `_first_item` guard so this
  change stands alone; it intentionally duplicates the central helper
  proposed separately and can collapse onto it if both land.

**Live verification (Network 10.5.67, local API mode):** the old path 404s;
the new path round-trips — portal read, `portal_enabled=false` written and
confirmed by re-read, captive portal observed disabled while the guest
network's isolation stayed in force.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)

The old endpoint never worked, so no working caller can regress; the new
`portal_enabled` parameter is optional.

## Testing

- Unit tests rewritten against the real `setting/guest_access` payload
  shape (previous fixtures mirrored the imagined API): auth-mode
  disambiguation, secret stripping, `portal_enabled` round-trip,
  version-gated customization keys with `skipped_fields`, unechoed-PUT
  re-read.
- `pytest tests/unit/` and `pre-commit run --all-files` pass.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (CHANGELOG.md)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published (none — stands alone)

## AI Assistance

This PR was created with assistance from Claude Code.

**Human Review Status:** ✅ Reviewed and approved by @jpennin5
**Testing:** Full unit suite passing; endpoint behavior verified against a
live Network 10.5.67 controller.
**Security Review:** Completed — all fixture values are synthetic or from
documentation ranges; stored portal secrets (`x_*` keys) are stripped from
tool output.

### CODE REVIEW

File: src/tools/radius.py [hunk 1]
```diff
@@ -649,36 +656,106 @@ async def delete_radius_account(
...
+    auth = section.get("auth", "none")
+    if auth == "hotspot":
+        if section.get("password_enabled"):
+            auth_method = "password"
+        elif section.get("voucher_enabled"):
+            auth_method = "voucher"
+        elif section.get("radius_enabled"):
+            auth_method = "radius"
+        else:
+            auth_method = "hotspot"
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/radius.py:701]: Considered, but remapping to "none" would be unsafe here: "none" is a *configurable* value meaning open access, so a caller that reads the remapped state and writes it back would actively open the portal. The ambiguous state stays reported faithfully as the output-only value "hotspot" — documented at the point it's produced in eb57456 — with the raw section alongside as ground truth.

File: src/tools/radius.py [hunk 2]
```diff
@@ -649,36 +656,106 @@ async def delete_radius_account(
...
+def _first_item(response: Any) -> dict[str, Any]:
+    """First object from a controller response, ``{}`` when nothing came back.
+
+    ``response.get("data", [{}])[0]`` does not survive an accepted-but-unechoed
+    write: the default only applies when the key is absent, not when the list
+    is empty.
+    """
+    items = response if isinstance(response, list) else response.get("data", [])
+    if not isinstance(items, list) or not items:
+        return {}
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/radius.py]: Fixed in eb57456 — non-dict replies now yield `{}` instead of raising.

File: src/tools/radius.py [hunk 3]
```diff
@@ -707,78 +793,105 @@ async def configure_guest_portal(
...
+                payload["auth"] = "hotspot"
+                for method, flag in _HOTSPOT_METHOD_FLAGS.items():
+                    payload[flag] = method == auth_method
+            elif auth_method == "external":
+                payload["auth"] = "custom"
+            else:
+                payload["auth"] = "none"
         if password is not None:
-            payload["password"] = password
+            payload["x_password"] = password
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/radius.py:884]: Added in eb57456: a password with a non-password method is rejected; switching to password auth with nothing provided and nothing stored is rejected; and a password without the method switch is rejected with guidance to pass auth_method='password'.

File: tests/unit/tools/test_radius_tools.py [hunk 4]
```diff
@@ -413,75 +426,140 @@ async def test_get_guest_portal_config_success(mock_settings):
...
+    updated = _guest_access_section(portal_enabled=False)
+
+    with patch("src.tools.radius.UniFiClient") as mock_client_class:
+        mock_client = AsyncMock()
+        mock_client.is_authenticated = False
+        mock_client.authenticate = AsyncMock()
+        mock_client.get = AsyncMock(return_value={"data": [current]})
+        mock_client.put = AsyncMock(return_value={"data": [updated]})
+        mock_client.__aenter__ = AsyncMock(return_value=mock_client)
+        mock_client.__aexit__ = AsyncMock()
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply tests/unit/tools/test_radius_tools.py]: Fixed in eb57456 for all 58 occurrences in the file, not just these five — and the change immediately earned its keep: two existing account tests were passing over a swallowed pydantic ValidationError (fixtures missing the required `site_id`). Their fixtures are corrected and the failures now surface.

**Comment by enuno** [comment]
### Summary
Request changes. The PR is not yet merge-ready because the current status checks are failing.

### Critical findings
- Merge state: \ / \
- Failing check(s): Mermaid Diagram Sync Assistant
- The changed tooling surface should be revalidated after the failing check is resolved.

### Suggestions
1. Resolve the failing status check(s) and rerun CI.
2. Reconfirm the changed tooling behavior against the intended controller/fixture shapes.
3. Ensure regression tests cover the original failure mode and edge cases introduced here.
4. Update docs if any request/response or error semantics changed.

### Verification checklist for the author
- [ ] Required checks are passing
- [ ] Regression tests cover the changed path
- [ ] Lint / format / type checks pass locally
- [ ] Documentation is updated where user-facing behavior changed
- [ ] No secrets, credentials, or controller-identifying data were introduced

**Review by enuno — CHANGES_REQUESTED** [review]
Merge blocked until the failing status checks are resolved and the tooling changes are revalidated.

**Comment by jpennin5** [comment]
The failing check is the `Mermaid Diagram Sync Assistant` GitHub App, not a workflow in this repository. Its check-run output on this PR reads: "Webhook Processing Failed — An unexpected error occurred while processing the webhook. Please try pushing a new commit or check your repository configuration." That processing happens on the app's side; pushing new commits to these PRs has re-triggered the check with the same result, so it is not reachable from a fork.

This is the loop documented in #112: the app has errored on every fork PR since late June, fork CI runs sit at `action_required` awaiting maintainer approval, and the daily automated review then reads the app error as a failing required check and requests changes.

Every check that can conclude on a fork PR is green here (both Socket Security checks). The remaining steps are repo-side — approving the CI runs and either repairing/removing the app integration or unrequiring its check. #117 merged with the identical check state, so the merge path exists. Happy to rebase or adjust anything code-side the moment CI can actually run.

**Comment by enuno** [comment]
Summary: request changes

Critical findings:
- Merge conflict / dirty merge state blocks safe review.

Suggestions:
1. Rebase onto main and resolve conflicts in: CHANGELOG.md, src/a2a/playbooks/guest_wifi_setup.py, src/models/radius.py, src/tools/radius.py, tests/unit/tools/test_radius_tools.py.
2. Re-run the touched test suite and pre-commit.
3. Re-check the feature behavior called out in the PR description.

Verification checklist:
- [ ] Rebasing completed cleanly
- [ ] Tests pass
- [ ] Lint/type checks pass
- [ ] Documentation remains aligned with implementation

**Review by enuno — CHANGES_REQUESTED** [review]
Automated review: merge conflicts/dirtiness detected (CONFLICTING/DIRTY). Please rebase and resolve before resubmitting.

**Comment by jpennin5** [comment]
Rebased onto current `main` (2c78bba) — the conflict was in `src/tools/radius.py` (with #123's hotspot-package work) and CHANGELOG, and it is resolved.

Resolution was gated rather than eyeballed: every CHANGELOG bullet already on `main` is verified still present (union, not overwrite), the sibling work merged into the shared files is verified intact, the full unit suite passes, and pre-commit passes for every file the PR touches.

Codecov's 74% patch report is answered too: the radius/hotspot/external/none translation arms, the missing-section error, the external and none write payloads, and the unwrap helper's bare-list and scalar paths are now covered.

GitHub now reports this PR mergeable.

**Comment by enuno** [comment]
## Summary
Request changes.

## Critical findings
- CI is not green on this PR.
- Failing checks: Mermaid Diagram Sync Assistant; Pre-commit Hooks.
- Touched scope: CHANGELOG.md, src/a2a, src/models, src/tools.

## Suggestions
1. Fix the failing checks on the current branch and rerun them until the gate is green.
2. Re-verify the changed code paths and tests for the touched scope above.
3. If any failing security or lint checks are expected, document the rationale and follow-up needed.

## Verification checklist
- [ ] `pre-commit run --all-files` passes
- [ ] Security/lint checks are green or explicitly justified
- [ ] Relevant unit tests cover the changed behavior
- [ ] No unintended API or documentation regressions

**Review by enuno — CHANGES_REQUESTED** [review]
CI is red on this PR (Mermaid Diagram Sync Assistant; Pre-commit Hooks). Please resolve the failing checks before merge.
